openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -518,34 +523,88 @@ class SequenceMetadata(DeveloperErrorViewMixin, APIView):
         SessionAuthenticationAllowInactiveUser,
     )
 
-    def get(self, request, usage_key_string, *args, **kwargs):  # lint-amnesty, pylint: disable=unused-argument
+    def get(self, request, usage_key_string, **kwargs):
         """
         Return response to a GET request.
         """
         try:
             usage_key = UsageKey.from_string(usage_key_string)
-        except InvalidKeyError:
-            raise NotFound(f"Invalid usage key: '{usage_key_string}'.")  # lint-amnesty, pylint: disable=raise-missing-from
+            learning_sequence = get_learning_sequence(usage_key)
+        except (InvalidKeyError, LearningSequenceData.DoesNotExist) as exc:
```

Comment:
Can you please separate the logged error if we have `LearningSequenceData.DoesNotExist`, and have it not fail altogether when that happens?

Reply:
Will do.